### PR TITLE
Squash migrations and add dbdocs GitHub action

### PR DESCRIPTION
**Pull request by frankhereford**
# Associated issues

This PR looks to close https://github.com/cityofaustin/atd-data-tech/issues/18793 and to close https://github.com/cityofaustin/atd-data-tech/issues/14059. 

# Testing

* https://dbdocs.io/transportation.data/Vision-Zero-Staging
* Spin up your local database with `replicate-db`. Still work?
* If you want, step through the steps found [here](https://github.com/cityofaustin/vision-zero/blob/add-schema-spy-github-action/.github/workflows/dbdocs.yml#L49-L54) if you want to simulate what the github action is going to do.

# Note
As John noted, you'll need to "apply" (but skip) this migration to get a production replication working with this new migration stream.

```bash
hasura migrate apply --skip-execution --database-name default --version 1728069606075
```

---

#### Ship list

- [ ] Check migrations for any conflicts with latest migrations in `main` branch
- [ ] Confirm Hasura role permissions for necessary access
- [ ] Code reviewed
- [ ] Product manager approved

### CODE REVIEW

File: vision-zero
Comment (frankhereford) [inline vision-zero:1]: Changes in here are related to the compose file consolidation.

File: .github/workflows/schema_spy.yml
Comment (frankhereford) [inline .github/workflows/schema_spy.yml:116]: This will come back in if we decide to keep schema spy after we identify and enhance the IAM account used in the GH secrets.

File: .github/workflows/schema_spy.yml
Comment (frankhereford) [inline .github/workflows/schema_spy.yml:7]: These will replace the following line before going into production. This allows for this to be tested by pushing on this branch.

File: database/migrations/default/1728069606075_schema/up.sql
Comment (johnclary) [inline database/migrations/default/1728069606075_schema/up.sql:1]: super minor, but would you mind naming this migration `_init` ? it helps me to quickly ignore it when it turns up in search results.

File: database/migrations/default/1728486708241_populate-lookup-tables/up.sql
Comment (johnclary) [inline database/migrations/default/1728486708241_populate-lookup-tables/up.sql:1]: thanks for cooking up these two post-init migrations. these are looking great, but i kind of want to carve them out of this PR and leave them for https://github.com/cityofaustin/atd-data-tech/issues/18796.

As I see it, that feature won't be fully baked until we create seeds, update the docs, and maybe polish the `vision-zero` helper script.

File: .github/workflows/schema_spy.yml
Comment (johnclary) [inline .github/workflows/schema_spy.yml:138]: Would you mind adding a section to the bottom of the [database readme](https://github.com/cityofaustin/vision-zero/blob/main/database/README.md) that summarizes our use of `dbdocs.io` and any breadcrumbs to 1password, etc?

**Review by johnclary — CHANGES_REQUESTED** [review]
wow—thanks, Frank, for getting us a fresh start on the db state!

the docs sites look great. i think dbdocs.io will be a nice upgrade over schemaspy, especially considering that it does a better job handling multiple DB schema.

i feel pretty strongly that we should **not** maintain both schemaspy and dbdocs. if others agree that we go with dbdocs, i'm cool with stripping out the schemaspy content from here.

i did want to mention for others that in order to start my local copy of prod, i had to apply these migrations with the `--skip-execution` flag. this will only be necessary until we do the same against prod.

E.g., 

```shell 
hasura migrate apply --skip-execution --version 1728488607807
```

File: .gitignore
Comment (frankhereford) [reply .gitignore:95]: 😨 Good gracious John, thank you for catching that!! 

I had added those because during testing, right in the root of the repo, I had `npm install`'d the `dbdocs` tooling, which created those package files and a `node_modules` directory. It was causing my VSCode to go bonkers with all the files it created, so I stuck them in the `.gitignore` file -- never even considered the utter mayhem it would create for the apps in the repo.

File: database/.gitignore
Comment (frankhereford) [reply database/.gitignore]: That would have been cool yea. It'd have given us some sort of diff from one version to the other, but interestingly enough, dbdocs is going to do that same thing for us.

File: database/migrations/default/1728069606075_schema/up.sql
Comment (frankhereford) [reply database/migrations/default/1728069606075_schema/up.sql:1]: Absolutely - great idea!

File: .github/workflows/schema_spy.yml
Comment (johnclary) [reply .github/workflows/schema_spy.yml]: oh, of course—migrations! glad to hear that ended up being helpful 🙌

File: database/migrations/default/1728486708241_populate-lookup-tables/up.sql
Comment (frankhereford) [reply database/migrations/default/1728486708241_populate-lookup-tables/up.sql:1]: Sure thing; I zapped them out of there. Good call on not delivering something half baked.

Comment (frankhereford) [reply database/migrations/default/1728486708241_populate-lookup-tables/up.sql:1]: You bet - they are removed. Will this prevent us from writing any migrations that adjust things like lookup tables until we get that work back in? Not a big deal, just a thought.

File: .github/workflows/schema_spy.yml
Comment (frankhereford) [reply .github/workflows/schema_spy.yml]: Great idea - it's done.

**Comment by frankhereford** [comment]
@johnclary I think this is ready to come back to you - thank you for all this amazing feedback.

File: .github/workflows/dbdocs.yml
Comment (chiaberry) [inline .github/workflows/dbdocs.yml:6]: should it also run on every push to production?

File: database/README.md
Comment (chiaberry) [inline database/README.md:412]: small typo teh/the

File: .github/workflows/dbdocs.yml
Comment (chiaberry) [reply .github/workflows/dbdocs.yml:6]: oh and I guess this line would be removed after testing is done/ before it is merged to main/prod?

**Comment by chiaberry** [comment]
This could be what yall were mentioning yesterday, but when I replicated prod this morning and tried to apply the migrations to spin up the vze, I got an error
```
    "error": {
      "description": null,
      "exec_status": "FatalError",
      "hint": null,
      "message": "schema \"lookups\" already exists",
      "status_code": "42P06"
    },
```

File: .github/workflows/dbdocs.yml
Comment (frankhereford) [reply .github/workflows/dbdocs.yml:6]: Yea, and I think it's good time to do that now. Thank you for mentioning this! 089c4ad1d4c0f390e681a1e87952983f53278d76

File: database/README.md
Comment (frankhereford) [reply database/README.md]: Thanks - great catch! f552d2f8ea3bb7374907a59688dde4c84f44db71

**Comment by frankhereford** [comment]
@chiaberry wrote:

> This could be what yall were mentioning yesterday, but when I replicated prod this morning and tried to apply the migrations to spin up the vze, I got an error

Yes - this is expected until we get this into production because the production DB still is thinking with the old migrations, and this resets all that into a new set of migrations. 

I believe during launch, we'll need to use a `hasura` client to inform the production DB that we'd like it to forget all migrations prior to merger so our CI doesn't run into this same error you did. 

It's helpful that you surfaced this; it's good to think about the ins-and-outs of this getting into production because it's kinda strange.

File: vision-zero
Comment (mddilley) [inline vision-zero:135]: Should this arg be removed from the arg parser and also from the local dev docs now that this Docker Compose file is removed?

File: docker-compose-github-dbdocs-action.yml
Comment (mddilley) [inline docker-compose-github-dbdocs-action.yml:3]: Thinking out loud but it might be worth adding this to the launch checklist so we remember to bump the Hasura version here too

**Review by mddilley — COMMENTED** [review]
Awesome - I know it took some whittling to get to the final GH action workflow, but you made it look easy in that file! I tested the fresh migration ✨, and I also mimicked the GH action and I see a database with all schema and no data. Looks so good! 🙌

I left a few questions - one or two that might need changes, but I'll get an approval on this asap.

File: docker-compose-github-dbdocs-action.yml
Comment (johnclary) [reply docker-compose-github-dbdocs-action.yml:3]: great call—i'll update the launch checklist

**Review by johnclary — COMMENTED** [review]
thanks @frankhereford for those changes. this looks great and i am ready to approve.

one last thing, i just wanted to get your thoughts on the launch plan. would you mind writing up those steps here or in one of the linked issues?

File: .github/workflows/dbdocs.yml
Comment (frankhereford) [reply .github/workflows/dbdocs.yml:64]: Howdy - great question. Yes, on first run in production via a the github workflow, we'll get our production one created. It's entirely based on the `$PROJECT_NAME` environment variable, and it "upserts" the documentation, so it'll create a new set if needed.

File: vision-zero
Comment (frankhereford) [reply vision-zero:135]: Yes! Thank you - great catch! c157b8656eaf421e53eea7a91a7d1b6c22c64f78

**Review by mddilley — APPROVED** [review]
🔍 🚢  🙌

**Comment by frankhereford** [comment]
@johnclary wrote:

> would you mind writing up those steps here or in one of the linked issues?

Sure -- here's my take. I'm curious if this lines up with what you had envisioned.

# Launch Steps for Reset of VZ Migrations

## Merge Freeze 

* We want to get the production database in sync with the migrations found in `main`
* So, any open PRs that contain migrations need to get merged into production
* And at that point, we call a merge freeze to keep it in sync.

## Update PR with `pg_dump` of production into initial migration

* If any migrations have made it into `main` since the initial migration was created, then it needs to be recreated with the following:

```bash
docker exec -it vision-zero-postgis-1 pg_dump atd_vz_data -U visionzero --schema-only --exclude-schema=hdb_catalog --no-owner > ./database/migrations/default/1728069606075_init/up.sql
```
* Commit this into the PR

## Backup the DB

* RDS snapshot

## Disable the workflow which applies migrations

* This is done via the GitHub web UI under the '...' hamburger when viewing a workflow

## Setup local `hasura` client for production and prepare target DB

* Setup the bastion as a place to run the `hasura` client with a pair of configs pointing to the staging and production databases.
* `hasura migrate delete` to empty out the production databases understanding of the migrations on both of them

## Merge the PR into `main`

* This will put the new migrations in place in the code

## Bring the staging DB into sync with the migrations in `main`

* With the staging config in place:
* `hasura migrate apply --skip-execution --database-name default --version 1728069606075`

## Half way there! 😅 Prepare a PR from `main` to `production`

* Merge it, observing that the migration applying workflow remains disabled

## Bring the production DB into sync with the migrations in `main`

* With the production config in place:
* `hasura migrate apply --skip-execution --database-name default --version 1728069606075`

## Re-enable the workflow which applies migrations

* Done in the same place -- github web UI.

**Review by johnclary — APPROVED** [review]
awesome—that all sounds great to me! TIL you can manually disable Github action workflows. one minor detail, in case it's helpful: i tested the step to delete migrations and learned that you can do `hasura migrate delete --all --server` to prevent Hasura from deleting your local migration files. that was the trick i needed after i wiped out my own migrations directory on first go.

in terms of when to move forward—i think we should push ahead, the sooner the better!

**Pull request merged**
